bridges/otelzap/core.go

```diff
@@ -108,10 +110,20 @@ func (o *Core) Enabled(level zapcore.Level) bool {
 	return o.logger.Enabled(context.Background(), r)
 }
 
-// TODO
 // With adds structured context to the Core.
 func (o *Core) With(fields []zapcore.Field) zapcore.Core {
-	return o
+	clone := o.clone()
+	if len(fields) > 0 {
+		clone.attr = append(clone.attr, convertField(fields)...)
+	}
+	return clone
```

Comment:
nit: using the same name twice can make it confusing for the reader. Can we refactor e.g. to

```suggestion
	cloned := o.clone()
	if len(fields) > 0 {
		cloned.attr = append(cloned.attr, convertField(fields)...)
	}
	return cloned
```